znft/ethereum_wallet.go

```diff
@@ -0,0 +1,93 @@
+package znft
+
+import (
+	"context"
+	"math/big"
+	"path"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/pkg/errors"
+)
+
+func (conf *Configuration) CreateEthClient() (*ethclient.Client, error) {
+	client, err := ethclient.Dial(conf.EthereumNodeURL)
+	if err != nil {
+		Logger.Error(err)
+	}
+	return client, err
+}
+
+func (conf *Configuration) createSignedTransactionFromKeyStore() *bind.TransactOpts {
+	var (
+		signerAddress = common.HexToAddress(conf.WalletAddress)
+		password      = conf.VaultPassword
+		value         = conf.Value
+	)
+
+	client, err := conf.CreateEthClient()
+	if err != nil {
+		Logger.Fatal(errors.Wrap(err, "failed to create ethereum client"))
```

Comment:
Let's return or exit with error explicitly (all the places below as well) as if any chance the Logger level is < FATAL, the logger would do nothing, therefore the following code will be executed unexpectedly, and could be serious bug.

Reply:
Fixed in 6a4ac251daacdd8e99308f388575ea47c0b5bd5b